lib/indexer/records.js

```diff
@@ -0,0 +1,152 @@
+/*!
+ * records.js - indexer records
+ * Copyright (c) 2018, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+/**
+ * @module lib/records
+ */
+
+const bio = require('bufio');
+const util = require('../utils/util');
+const consensus = require('../protocol/consensus');
+
+/**
+ * Block Meta
+ * @alias module:indexer.BlockMeta
+ */
+
+class BlockMeta {
+  /**
+   * Create block meta.
+   * @constructor
+   * @param {Hash} hash
+   * @param {Number} height
+   */
+
+  constructor(hash, height) {
+    this.hash = hash || consensus.NULL_HASH;
+    this.height = height != null ? height : -1;
+  }
+
+  /**
+   * Clone the block.
+   * @returns {BlockMeta}
+   */
+
+  clone() {
+    return new this.constructor(this.hash, this.height);
+  }
+
+  /**
+   * Get block meta hash as a buffer.
+   * @returns {Buffer}
+   */
+
+  toHash() {
+    return Buffer.from(this.hash, 'hex');
+  }
+
+  /**
+   * Instantiate block meta from chain entry.
+   * @private
+   * @param {IndexEntry} entry
+   */
+
+  fromEntry(entry) {
+    this.hash = entry.hash;
+    this.height = entry.height;
+    return this;
+  }
+
+  /**
+   * Instantiate block meta from json object.
+   * @private
+   * @param {Object} json
+   */
+
+  fromJSON(json) {
+    this.hash = util.revHex(json.hash);
```

Comment:
`util.revHex` expects a buffer, but a `json.hash` is generally a string

Reply:
These methods are not actually used, and can be removed.